File: cli/testdata/python/quantum_families_usage.py
"""Tier-3 quantum-vulnerable family fixtures (issue #33)."""

from py_ecc.bls import G2ProofOfPossession as bls
from coincurve import PrivateKey


def sign_bls(sk, message):
    # EXPECTED: BLS12-381 | signature | bls | info | quantum-vulnerable
    sig = bls.Sign(sk, message)
    return sig


def verify_bls(pk, message, sig):
    # EXPECTED: BLS12-381 | signature | bls | info | quantum-vulnerable
    return bls.Verify(pk, message, sig)


def aggregate_bls(sigs):
    # EXPECTED: BLS12-381 | signature | bls | info | quantum-vulnerable
    return bls.Aggregate(sigs)


def sign_taproot(key: PrivateKey, msg: bytes) -> bytes:
    # EXPECTED: Schnorr | signature | schnorr | info | quantum-vulnerable
    return key.sign_schnorr(msg)


def verify_taproot(pub, sig, msg) -> bool:
    # EXPECTED: Schnorr | signature | schnorr | info | quantum-vulnerable
    return pub.verify_schnorr(sig, msg)

File: cli/testdata/javascript/quantum_families_usage.js
// Tier-3 quantum-vulnerable family fixtures (issue #33).
const { schnorr } = require('@noble/secp256k1');
const bls = require('@noble/bls12-381');

async function signTaproot(priv, msg) {
  // EXPECTED: Schnorr | signature | schnorr | info | quantum-vulnerable
  const sig = await schnorr.sign(msg, priv);
  return sig;
}

async function verifyTaproot(sig, msg, pub) {
  // EXPECTED: Schnorr | signature | schnorr | info | quantum-vulnerable
  return schnorr.verify(sig, msg, pub);
}

async function signBLS(priv, msg) {
  // EXPECTED: BLS12-381 | signature | bls | info | quantum-vulnerable
  const sig = await bls.sign(msg, priv);
  return sig;
}

async function aggregateBLS(sigs) {
  // EXPECTED: BLS12-381 | signature | bls | info | quantum-vulnerable
  return bls.aggregateSignatures(sigs);
}

module.exports = { signTaproot, verifyTaproot, signBLS, aggregateBLS };
